src/closure.rs

```diff
@@ -0,0 +1,136 @@
+// Copyright 2013-2017, The Gtk-rs Project Developers.
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+// TODO: support marshaller.
```

Comment:
I would like to have something like `GCClosure`, i.e. being able to use a function with Rust types.
I believe it should be done in another pull request since it looks non-trivial to do something both safe and efficient.